### PR TITLE
fix memoryview compare

### PR DESCRIPTION
**Pull request by hyperbora**

### CODE REVIEW

**Comment by youknowone** [comment]
@qingshi163 I think you are one of the pro of memoryview. Could you review this PR please?

File: vm/src/builtins/bytes.rs [hunk 1]
```diff
@@ -531,6 +531,13 @@ impl Comparable for PyBytes {
...
+        } else if other.isinstance(&vm.ctx.types.memoryview_type) {
+            return Err(vm.new_type_error(format!(
+                "'{}' not supported between instances of '{}' and '{}'",
+                op.operator_token(),
+                zelf.class().name,
+                other.class().name
+            )));
         } else {
             zelf.inner.cmp(other, op, vm)
         })
```
Comment (qingshi163) [inline vm/src/builtins/bytes.rs:543]: The CPython's logic is very strange here, memoryview could do == or != here with bytes, but raise type error when doing <, >, <=, >=. and bytearray just accept everything.
This code does not match the cpython's logic for the following compare:
```
bytes (==,!=) memoryview
memoryview (<,<=,>,>=)  bytearray
```

**Comment by hyperbora** [comment]
Thank you for your review. I'm not sure if I edited it as you said.

**Comment by qingshi163** [comment]
I don't know any reason CPython has a back door for ord compare between memoryview and bytearray(not bytes). Should we follow? @coolreader18 
Otherwise this code fix at lease some of the case.

**Comment by coolreader18** [comment]
Huh, that's really weird behavior. I think if there's no test cases that test for that (or even if there are, honestly) it's fine to ignore that.

**Review by coolreader18 — APPROVED** [review]
Thank you for contributing!

**Pull request merged**